pkg/ccl/backupccl/backup_planning.go

```diff
@@ -255,43 +238,18 @@ func getLogicallyMergedTableSpans(
 		// interleaved contains the tableID/indexID of the furthest ancestor in
 		// the interleaved chain.
 		if lhsIndex.IsInterleaved() || rhsIndex.IsInterleaved() {
+			mergedSpan.EndKey = lhsSpan.EndKey
```

Comment:
This looked wrong to me for a second since lhsIndex.Interleaved() might have been true and thus shouldn't be merged.  But the only time lhsIndex.IsInterleaved() should be true would be if it already the only thing in mergedSpans since a lhsIndex was previously a rhsIndex (or the first index). 

Also, whoa, doesn't this imply this function was rather wrong for cases with interleaved indexes? If so, might be worth mentioning in the commit message.